src/persist/src/location.rs

```diff
@@ -285,6 +320,34 @@ pub struct VersionedData {
     pub data: Vec<u8>,
 }
 
+impl<T: Codec> From<(SeqNo, &T)> for VersionedData {
```

Comment:
Code looks cleaner with these `From` impls! 🙌

Reply:
I wasn't entirely sold on this yet. glad to see that you seem to have a clearer opinion :D